src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -0,0 +1,136 @@
+package uk.gov.dwp.dataworks.services
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
+import software.amazon.awssdk.awscore.AwsRequest
+import software.amazon.awssdk.core.SdkRequest
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.ecs.EcsClient
+import software.amazon.awssdk.services.ecs.model.*
+import software.amazon.awssdk.services.ecs.model.LoadBalancer
+import software.amazon.awssdk.services.elasticloadbalancingv2.ElasticLoadBalancingV2Client
+import software.amazon.awssdk.services.elasticloadbalancingv2.model.*
+
+import java.net.http.HttpRequest
+
+@Service
+class TaskDeploymentService {
+    @Autowired
+    lateinit var credentialsService: CredentialsService
+
+    val awsRegion: Region = kotlin.runCatching { Region.of(System.getenv(ConfigKey.AWS_REGION.toString()))}.getOrDefault(Region.EU_WEST_2)
+
+    private fun createService (ecs_cluster_name: String, user_name: String, ecsClient: EcsClient, targetGroupArn: String) {
+
+        val alb: LoadBalancer = LoadBalancer.builder().targetGroupArn(targetGroupArn).containerPort(443).build()
+
+        val serviceBuilder = CreateServiceRequest.builder().cluster(ecs_cluster_name).loadBalancers(alb).serviceName("${user_name}_test").taskDefinition("mhf_sample_task").loadBalancers(alb).desiredCount(1).build()
+        println("Creating Service...")
+
+        try {
+            val service = ecsClient.createService(serviceBuilder)
+            println("service.responseMetadata = ${service.responseMetadata()}")
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+    }
+
+    fun getPriorityVal (rulesResponse : DescribeRulesResponse) : Int {
+
+        var consecVals : Int = 1
+        for (i in rulesResponse.rules()) {
+            if (i.priority() != "default") {
+                var intVal = Integer.parseInt(i.priority())
+                if (intVal == consecVals) consecVals = consecVals + 1
+            }
+        }
+        if (consecVals >= 1000) throw Exception("The upper limit of 1000 rules has been reached for this load balancer.")
+        return consecVals;
+    }
+
+    fun taskDefinitionWithOverride(ecs_cluster_name: String, emr_cluster_host_name: String, albName :String, user_name: String, jupyterCpu : Int=512, jupyterMemory: Int = 512) {
+
+        val credentials: AwsCredentialsProvider = credentialsService.getDefaultCredentialsProvider()
+
+        val ecsClient: EcsClient = EcsClient.builder().credentialsProvider(credentials).region(awsRegion).build()
+        val albClient: ElasticLoadBalancingV2Client = ElasticLoadBalancingV2Client.builder().credentialsProvider(credentials).region(awsRegion).build()
+
+        println("Getting alb information...")
+
+        val albRequest = DescribeLoadBalancersRequest.builder().names("orchestration-service-lb").build()
```

Comment:
This probably shouldn't be hardcoded here. I might suggest putting the names as an env var.